pkg/client/postrenderer.go

```diff
@@ -0,0 +1,132 @@
+package client
+
+import (
+	"bytes"
+
+	sdkhandler "github.com/operator-framework/operator-lib/handler"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/kube"
+	"helm.sh/helm/v3/pkg/postrender"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/cli-runtime/pkg/resource"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+
+	"github.com/operator-framework/helm-operator-plugins/internal/sdk/controllerutil"
+	"github.com/operator-framework/helm-operator-plugins/pkg/manifestutil"
+)
+
+func WithInstallPostRenderer(pr postrender.PostRenderer) InstallOption {
+	return func(i *action.Install) error {
+		i.PostRenderer = pr
+		return nil
+	}
+}
+
+func WithExtraInstallPostRenderer(pr postrender.PostRenderer) InstallOption {
+	return func(a *action.Install) error {
+		a.PostRenderer = addExtraPostRenderer(a.PostRenderer, pr)
+		return nil
+	}
+}
+
+func WithUpgradePostRenderer(pr postrender.PostRenderer) UpgradeOption {
+	return func(a *action.Upgrade) error {
+		a.PostRenderer = pr
+		return nil
+	}
+}
+
+func WithExtraUpgradePostRenderer(pr postrender.PostRenderer) UpgradeOption {
+	return func(a *action.Upgrade) error {
+		a.PostRenderer = addExtraPostRenderer(a.PostRenderer, pr)
+		return nil
+	}
+}
+
+func addExtraPostRenderer(pr postrender.PostRenderer, extra postrender.PostRenderer) postrender.PostRenderer {
+	if pr == nil {
+		return extra
+	}
+	return chainedPostRenderer{pr, extra}
```

Comment:
This ends up being a potentially recursive sort of thing. `pr` might itself be a `chainedPostRenderer`

Reply:
Well, if `chainedPostRenderer` was a struct type, yes. But since it's a slice type, then after multiple calls eventually you just end up with a single long slice, right?

Reply:
`chainedPostRenderer` is a slice of `postrender.PostRenderer`. The way this is setup now, we always return a wrapper `chainedPostRenderer` with exactly length 2. It just so happens that `pr` might already be a `chainedPostRenderer`, and if so, it is also of length 2.

So after two appends, you'd end up with:
```go
out := chainedPostRender{
	chainedPostRender{
		defaultPostRenderer,
		append1,
	},
	append2,
}
```

Reply:
[Right](https://go.dev/play/p/j8ZTG8wy2oi), we get a slice whose first element is a slice. The underlying type of `pr` does not matter here...

Reply:
So, should we do anything about this? If user stacks up four renderers, we end up with this leaning tree of two-element `ChainedPostRenderers`, and the `%d` on line 98 is going to be somewhat misleading/confusing - they will get a sequence of error messages all mentioning a combination of 1 and 0...
Perhaps use `append` if `pr` successfully type-asserts as a chainedPostRenderer?

Reply:
> the %d on line 98 is going to be somewhat misleading/confusing
good point!

i had the type assertion thing in an early draft and dropped it for simplicity, but it does seems worthwhile, so i'll add that back in. thanks!